test/helpers/kubectl.go

```diff
@@ -618,6 +671,98 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
 	return "", nil
 }
 
+// CiliumExportInfo runs a subroutine in the background, that exports the
+// Cilium memory usage and the given commands to the prometheus server. The
+// function also receives a context to stop the function when is needed and
+// prefix to store to send the data to the database. This function gather the
+// info each 5 seconds.
+func (kub *Kubectl) CiliumExportInfo(ctx context.Context, prefix string, cmds map[string]string) {
+	tickerSleep := 5 * time.Second
+
+	getKey := func(metric string) string {
+		return fmt.Sprintf("%s_%s", prefix, metric)
+	}
+
+	ExecReport := func() config.PrometheusMetrics {
+		result := config.PrometheusMetrics{}
+		memory, err := kub.CiliumReportMemory()
+		if err != nil {
+			kub.logger.WithError(err).Warn("cannot get memory report")
+		}
+
+		// Memory report first
+		for k, v := range memory {
+			result[getKey(k)] = fmt.Sprintf("%d", v)
+		}
+
+		for k, v := range cmds {
+			res := kub.Exec(v)
+			if !res.WasSuccessful() {
+				kub.logger.WithError(fmt.Errorf("%s", res.CombineOutput())).Warnf(
+					"cannot export metric '%s' with cmd '%s'", k, v)
+			}
+			result[getKey(k)] = res.SingleOut()
+		}
+		return result
+	}
+
+	ticker := time.NewTicker(tickerSleep)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				data := ExecReport()
+				err := config.PushInfo(&data)
+				if err != nil {
+					kub.logger.WithError(err).Warn("cannot push info to prometheus")
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// CiliumReportMemory runs kubectl top function and returns a map with the
+// total, avg, max and min memory used by cilium pods. In case that heapster is
+// not installed it returns a error.
+func (kub *Kubectl) CiliumReportMemory() (map[string]int, error) {
+	result := map[string]int{
+		Total: 0,
+		Avg:   0,
+		Max:   0,
+		Min:   0,
+	}
+
+	cmd := fmt.Sprintf(`%s -n %s top pod -l %s | grep "^cilium" | awk '{print $1"="($3+0)}'`,
+		KubectlCmd, KubeSystemNamespace, ciliumLabel)
+	res := kub.Exec(cmd)
+
+	if !res.WasSuccessful() {
+		return nil, fmt.Errorf("cannot get pods memory: %s", res.CombineOutput())
```

Comment:
error is a bit vague. What about `cannot gather memory statistics for pods: %s`?